VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionThread.java

```diff
@@ -178,6 +178,8 @@ public void setRead(boolean read) {
         this.read = read;
         if (read) {
             unreadCommentCount = 0;
+        } else {
+            unreadCommentCount = 1;
```

Comment:
This logic is wrong. If we were actually marking the whole thread as unread (as the method name would suggest, although this is not currently supported by the API or the UI), then this should be `unreadCommentCount == commentCount`. However, what we're actually using it for here is to _increment_ the `unreadCommentCount`, which should probably be implemented in a separate method (which also sets `read` to `false` incidentally).